src/screens/Customers/CustomerMain.tsx

```diff
@@ -1,29 +1,124 @@
+      height: '33.83px',
+      fontFamily: 'Helvetica Neue',
+      fontStyle: 'normal',
+      fontWeight: theme.typography.h1.fontWeight,
+      fontSize: theme.typography.h1.fontSize,
+      lineHeight: '115%',
+      color: '#828282',
+      flexGrow: 1,
+      paddingRight: '20px',
+    },
+    headerWrapper: {
+      display: 'flex',
+      flexDirection: 'row',
+      width: '100%',
+    },
+    selectionHeader: {
+      display: 'flex',
+      flexDirection: 'column',
+      flexGrow: 1,
+    },
+    scrollDiv: {
+      marginLeft: 'auto',
+      marginRight: 'auto',
+      width: '100%',
+      height: '100%'
+    },
+  });
+
+interface UserProps {
+  classes: { title: string; headerWrapper: string; selectionHeader: string; scrollDiv: string };
+}
+
+function CustomerMain(props: RouteComponentProps & UserProps) {
   useEffect(() => {
     getCustomers();
   }, []);
 
-  const [customers, setCustomers] = useState<CustomerRecord[]>([]);
+  const { classes } = props;
+  const [filteredCustomers, setFilteredCustomers] = useState<CustomerRecord[]>([]);
+  const [fullCustomers, setFullCustomers] = useState<CustomerRecord[]>([]);
+  const [allCustomersTrie, setAllCustomersTrie] = useState<TrieTree<CustomerRecord>>(new TrieTree('name'));
+
+  const getCustomers = () => {
+    const siteData = store.getState().siteData.currentSite;
+    const allCustomers: CustomerRecord[] = siteData.customers;
+    setFilteredCustomers(allCustomers);
+    setFullCustomers(allCustomers);
+    const customersTrie: TrieTree<CustomerRecord> = new TrieTree('name');
+    customersTrie.addAll(allCustomers);
+    setAllCustomersTrie(customersTrie);
 
-  const getCustomers = async () => {
-    const customers: CustomerRecord[] = await getAllCustomers();
-    setCustomers(customers);
-  };
+  }
+
+  const getLatestReadingDate = (customer: CustomerRecord) => {
+    //depends if the meter readings list is sorted with earliest => latest
+    if (customer.meterReadings.length > 0) {
+      const latestMeterReading = customer.meterReadings[customer.meterReadings.length - 1]
+      const dateTime = Date.parse(latestMeterReading.date)
+      const readingDate = new Date(dateTime)
+      const month = readingDate.getMonth() + 1;
+      const day = readingDate.getDate();
+      const shortDate = month.toString() + '.' + day.toString();
+      return shortDate;
+    } else {
+      return 'No Readings'
+    }
+  }
+
+  const handleSearchChange = (e: any) => {
+    const searchVal = e.target.value.trim();
+    if (searchVal === '') {
+      setFilteredCustomers(fullCustomers);
+      return;
+    }
+    const customers = allCustomersTrie.get(searchVal);
+    setFilteredCustomers(customers);
+  }
 
   return (
-    <>
-      {customers.map((customer, index) => (
-        <Link key={index} to={{ pathname: `${props.match.url}/${customer.name}`, state: { customer: customer } }}>
-          <h3>{customer.name}</h3>
-        </Link>
-      ))}
-    </>
+    <BaseScreen>
```

Comment:
Add `rightIcon user` to match Figma.

Right now:
![image](https://user-images.githubusercontent.com/35501399/104109615-98580c80-5284-11eb-91b4-c5c74150008b.png)

Figma:
![image](https://user-images.githubusercontent.com/35501399/104109618-a1e17480-5284-11eb-8a66-b26d7ab86dae.png)